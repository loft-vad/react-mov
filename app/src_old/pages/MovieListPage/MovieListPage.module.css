header {
  background: url(./../../assets/movies-header-bg.jpg) no-repeat center center;
  overflow: hidden;
  background-size: 98%;
  height: 450px;
  margin-bottom: 5px;
}
header .headerWrapper {
  background-color: rgba(0, 0, 0, 0.7);
  backdrop-filter: blur(4px);
  height: 100%;
  padding: 20px 60px;
}
header .headerWrapper h1 {
  padding-bottom: 38px;
}
header .headerWrapper .logoSmall {
  color: var(--main);
}
header .headerWrapper .headerBar {
  display: flex;
  justify-content: space-between;
  padding-bottom: 35px;
}
header .headerWrapper > div:last-child {
  padding: 0 60px;
}

.content {
  padding: 0 60px;
  background: var(--black);
}

.moviesList {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 54px;
}

.searchFilter {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 24px;
}

.counter {
  padding: 40px 60px;
  font-size: 55px;
}
.counter button {
  color: #fff;
  padding: 15px 45px;
  font-size: 22px;
  background: rgba(255, 255, 255, 0.2);
  margin: 10px;
  border-radius: 3px;
}

/*# sourceMappingURL=MovieListPage.module.css.map */
