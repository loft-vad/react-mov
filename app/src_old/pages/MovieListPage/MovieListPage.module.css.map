{"version":3,"sourceRoot":"","sources":["MovieListPage.module.scss"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;EACA;EACA;;AAEA;EACE;EACA;EACA;EACA;;AAEA;EACE;;AAGF;EACE;;AAGF;EACE;EACA;EACA;;AAGF;EACE;;;AAKN;EACE;EACA;;;AAGF;EACE;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAIF;EACE;EACA;;AAEA;EACE;EACA;EACA;EACA;EACA;EACA","file":"MovieListPage.module.css"}