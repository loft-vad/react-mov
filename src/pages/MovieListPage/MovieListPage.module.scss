header {
  background: url(./../../assets/movies-header-bg.jpg) no-repeat center center;
  overflow: hidden;
  background-size: 98%;
  height: 450px;
  margin-bottom: 5px;

  .headerWrapper {
    background-color: rgba(0, 0, 0, 0.7);
    backdrop-filter: blur(4px);
    height: 100%;
    padding: 20px 60px;

    h1 {
      padding-bottom: 38px;
    }

    .logoSmall {
      color: var(--main);
    }

    .headerBar {
      display: flex;
      justify-content: space-between;
      padding-bottom: 35px;
    }

    &>div:last-child {
      padding: 0 60px;
    }
  }
}

.contentWrapper {
  max-width: 1200px;
  margin: 0 auto;
}

.content {
  padding: 0 60px;
  background: var(--black)
}

.moviesList {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 54px;
}

.searchFilter {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 24px;
}

// TEMP
.counter {
  padding: 40px 60px;
  font-size: 55px;

  button {
    color: #fff;
    padding: 15px 45px;
    font-size: 22px;
    background: rgba(255, 255, 255, 0.2);
    margin: 10px;
    border-radius: 3px;

  }
}