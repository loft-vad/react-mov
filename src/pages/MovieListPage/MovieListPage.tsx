import React, { useEffect, useState } from "react";
import styles from "./MovieListPage.module.scss";

import SearchForm from "../../components/SearchForm/SearchForm";
import GenreSelect from "../../components/GenreSelect/GenreSelect";
import MovieTile from "../../components/MovieTile/MovieTile";
import MovieDetails, { MovieFull } from "../../components/MovieDetails/MovieDetails";
import SortControl, { sortByValue } from "../../components/SortControl/SortControl";
import AddMovie from "../../components/common/AddMovieButton/AddMovieButton";

import { ReactComponent as LogoSmall } from "../../assets/logo-small.svg";

export const sortBy: sortByValue[] = [
  { id: 1, name: "Release Date" },
  { id: 2, name: "Title" },
];

interface MovieApiData {
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
  id: number;
}

const moviesApiUrl = "http://localhost:4000/movies?";

const MovieListPage: React.FC = () => {
  const [searchQuery, setSearchQuery] = useState("");
  const [sortCriterion, setSortCriterion] = useState(sortBy[0].name);
  const [activeGenre, setActiveGenre] = useState<string>("All");
  const [movieList, setMovieList] = useState<MovieFull[]>([]);
  const [selectedMovie, setSelectedMovie] = useState<MovieFull | null>(null);
  const [loading, setLoading] = useState(false);
  const [controller, setController] = useState(new AbortController());

  const genres: string[] = [
    "All",
    "Documentary",
    "Comedy",
    "Horror",
    "Crime",
    "Fantasy",
    "Adventure",
    "Science Fiction",
  ];

  const onSearch = (searchString: string) => {
    console.log("searchString ", searchString);
    setSearchQuery(searchString);
  };

  const handleGenreSelection = (name: string) => {
    console.log(name, " clicked");
    setActiveGenre(name);
  };

  const handleMovieTileClick = (movie: MovieFull) => {
    console.log("Movie tile clicked");
    setSelectedMovie(movie);
  };

  const handleSortByClick = (criterion: string) => {
    console.log("Sort By item cliecked: ", criterion);
    setSortCriterion(criterion);
  };

  const dataTransform = (movieApi: MovieApiData[]): MovieFull[] => {
    return movieApi.map(
      ({ id, genres, poster_path, title, release_date, vote_average, overview, runtime }) => {
        return {
          id,
          genres,
          imageUrl: poster_path,
          movieName: title,
          releaseYear: +release_date.slice(0, 4),
          rating: vote_average,
          duration: runtime,
          description: overview,
        };
      },
    );
  };

  useEffect(() => {
    controller.abort();
    const newController = new AbortController();
    setController(newController);
    setLoading(true);

    let apiUrl = `${moviesApiUrl}`;
    if (activeGenre !== "All") {
      apiUrl = apiUrl + `searchBy=genres&search=${activeGenre}`;
    }
    if (searchQuery !== "") {
      apiUrl = apiUrl + `searchBy=title&search=${searchQuery}`;
    }
    if (sortCriterion !== sortBy[0].name) {
      apiUrl = apiUrl + `sortBy=${sortCriterion.toLowerCase()}&sortOrder=asc`;
    }
    console.log("apiUrl: ", apiUrl);

    fetch(apiUrl, { signal: newController.signal })
      .then((response) => response.json())
      .then((data) => {
        setMovieList(dataTransform(data.data));
      })
      .catch((error) => {
        console.error("Error fetching movies:", error);
      })
      .finally(() => {
        setLoading(false);
      });
    return () => {
      newController.abort();
    };
  }, [searchQuery, sortCriterion, activeGenre]);

  return (
    <>
      {selectedMovie ? (
        <div>
          <MovieDetails movieData={selectedMovie} />
        </div>
      ) : (
        <header>
          <div className={styles.headerWrapper}>
            <div className={styles.headerBar}>
              <div className={styles.logoSmall}>
                <LogoSmall />
              </div>
              <AddMovie />
            </div>
            <div>
              <h1>Find Your Movie</h1>
              <SearchForm onSearch={onSearch} />
            </div>
          </div>
        </header>
      )}
      <div className={styles.content}>
        <div className={styles.searchFilter}>
          <GenreSelect genres={genres} onSelect={handleGenreSelection} />
          <SortControl selected={sortCriterion} values={sortBy} onSelect={handleSortByClick} />
        </div>
        <div className={styles.moviesList}>
          {movieList.map((movie) => (
            <MovieTile
              key={movie.id}
              movieData={movie}
              onClickHandler={() => handleMovieTileClick(movie)}
            />
          ))}
        </div>

        {loading ? <div>Loading...</div> : <div>Loaded</div>}
      </div>
    </>
  );
};

export default MovieListPage;
