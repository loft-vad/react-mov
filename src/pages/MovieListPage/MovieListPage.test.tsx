import React from "react";

import { render } from "@testing-library/react";
import MovieListPage from "./MovieListPage";

describe("MovieTile component", () => {
  it("should match snapshot", () => {
    const wrapper = render(<MovieListPage />);

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
});
