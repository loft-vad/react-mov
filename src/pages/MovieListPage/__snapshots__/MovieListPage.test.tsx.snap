// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile component should match snapshot 1`] = `
<header>
  <div
    class="headerWrapper"
  >
    <div
      class="headerBar"
    >
      <div
        class="logoSmall"
      >
        <svg>
          logo-small.svg
        </svg>
      </div>
      <button
        class="addMovieButton"
      >
        + Add Movie
      </button>
    </div>
    <div>
      <h1>
        Find Your Movie
      </h1>
      <div
        class="searchWrapper"
      >
        <input
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
        <button
          class="button"
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</header>
`;
